src/xs_driver.cpp

```diff
@@ -870,10 +903,18 @@ bool InterbotixDriverXS::retrieve_motor_configs(
     Gripper gripper;
     // load all info from the single_gripper node into the Griper struct, substituting the default
     //  values if not given the value
+    gripper.type = single_gripper["type"].as<std::string>("swing_arm");
```

Comment:
Make `"swing_arm"` and `"rack_and_pinion"` namespaced constant strings like what is done for mode and profile_type variables.